include/internal/yamerl_constr.hrl

```diff
@@ -57,6 +58,7 @@
     options              = []         :: [yamerl_constr_option()],
     ext_options          = []         :: [{term(), term()}],
     detailed_constr      = false      :: boolean(),
+    allow_duplicate_keys = false      :: boolean(),
```

Comment:
You don't need to modify the `#yamerl_conttr{}` record or the `yamerl_constr_option()` type.

Grep for the `str_node_as_binary` option to see an example. This option is used by `yamerl_node_str` and `yamerl_node_str_json` to ask for strings to be returned as binaries instead of strings.

Basically, options unknown to the parser or the constructor or stored in `ext_options` and can be used by `yamerl_node_*` modules.

Reply:
Ah nice! I already had the impression that I needed to change too much. Thanks for pointing me in the right direction.